### PR TITLE
[PRTL-2900] add “clear” function to search boxes

### PR DESCRIPTION
**Pull request by caravinci**
## Environment to be used in testing

- [ ] `prod`
- [x] `dev-oicr`
- [x] `qa` (which version?)

## Description of Changes
Adds clearing functionalities to the search input fields.

### CODE REVIEW

File: src/packages/@ncigdc/components/Aggregations/TermAggregation.js [hunk 1]
```diff
@@ -71,149 +71,148 @@ const TermAggregation = (props: TProps) => {
...
-              ...filter.content,
-              value: typeof filter.content.value === 'string'
+            .map(filter => ({
+              ...filter,
+              content: {
+                ...filter.content,
+                value: typeof filter.content.value === 'string'
                 ? filter.content.value.toLowerCase()
-                : filter.content.value.map(val => val.toLowerCase())
+                : filter.content.value.map(val => val && `${val}`.toLowerCase()),
```
Comment (yuanhengzhao) [inline src/packages/@ncigdc/components/Aggregations/TermAggregation.js]: What the type value can be? Is that ok to assume that the value is an array if it's not a string?

Comment (caravinci) [reply src/packages/@ncigdc/components/Aggregations/TermAggregation.js]: If the value is neither a string or an array, then we have an API issue.
But you're right, we could add an extra check here. Await amendment.

File: src/packages/@ncigdc/modern_components/BiospecimenCard/BioTree/index.js [hunk 2]
```diff
@@ -204,25 +203,12 @@ const EnhancedBranch = compose(
...
-      const {
-        setEmitterToken,
-        setExpanded,
-      } = this.props;
-      const token = Emitter.addListener('expand', toExpand => {
-        setExpanded(toExpand);
-      });
-
-      setEmitterToken(token);
-    },
```
Comment (yuanhengzhao) [inline src/packages/@ncigdc/modern_components/BiospecimenCard/BioTree/index.js]: No sure why this part is removed.

File: src/packages/@ncigdc/modern_components/SuggestionFacet/SuggestionFacet.js [hunk 3]
```diff
@@ -26,7 +25,6 @@ import styled from '@ncigdc/theme/styled';
 import { dropdown } from '@ncigdc/theme/mixins';
 import Link from '@ncigdc/components/Links/Link';
 import CheckCircleOIcon from '@ncigdc/theme/icons/CheckCircleOIcon';
-import { IRawQuery } from '@ncigdc/utils/uri/types';
```
Comment (yuanhengzhao) [inline src/packages/@ncigdc/modern_components/SuggestionFacet/SuggestionFacet.js]: This is used in 126 line. 
```suggestion
import { IRawQuery } from '@ncigdc/utils/uri/types';
```

File: src/packages/@ncigdc/utils/emitter.js [hunk 4]
```diff
@@ -1,6 +0,0 @@
-// @flow
-import { EventEmitter } from 'fbemitter';
-
-const emitter = new EventEmitter();
-
-export default emitter;
```
Comment (yuanhengzhao) [inline src/packages/@ncigdc/utils/emitter.js]: Same as above. Only want to know why it's being removed.

File: src/packages/@ncigdc/components/HomeSearch.js [hunk 5]
```diff
@@ -73,47 +84,77 @@ const HomeSearch = compose(
...
+          }}
+          />
+      )}
 
       <QuickSearchResults
+        isLoading={state.isLoading}
+        onActivateItem={selectItem}
+        onSelectItem={setFocusedItem}
+        query={state.query}
         results={state.results.map(
```
[the remark this answers is not in the document]
Comment (caravinci) [reply src/packages/@ncigdc/components/HomeSearch.js]: results is an array as default, from withSearch

File: src/packages/@ncigdc/modern_components/BiospecimenCard/BioTree/index.js (hunk 2 above)
Comment (caravinci) [reply src/packages/@ncigdc/modern_components/BiospecimenCard/BioTree/index.js]: it breaks the app, and isn't necessary after the BioTree state management reimplementation.

File: src/packages/@ncigdc/components/QuickSearch/QuickSearch.js [hunk 6]
```diff
@@ -169,11 +193,11 @@ const QuickSearch = ({
         results={state.fileHistoryResult
```
[the remark this answers is not in the document]
Comment (caravinci) [reply src/packages/@ncigdc/components/QuickSearch/QuickSearch.js]: fileHistoryResult is also an array as default, from withSearch

File: src/packages/@ncigdc/components/Aggregations/SuggestionFacet.js [hunk 7]
```diff
@@ -67,182 +66,168 @@ const StyledDropdownLink = styled(Link, {
...
-                          ...dropdown,
-                          marginTop: 0,
-                          top: '35px',
-                          width: '300px',
-                          wordBreak: 'break-word',
-                        }}
-                        >
-                        {(results || []).map(x => (
+                      >
+                      {results.length > 0
```
[the remark this answers is not in the document]
Comment (caravinci) [reply src/packages/@ncigdc/components/Aggregations/SuggestionFacet.js]: Results defaults to an array, as per line 78

File: src/packages/@ncigdc/modern_components/SuggestionFacet/SuggestionFacet.js [hunk 8]
```diff
@@ -26,7 +25,6 @@ import styled from '@ncigdc/theme/styled';
 import { dropdown } from '@ncigdc/theme/mixins';
 import Link from '@ncigdc/components/Links/Link';
 import CheckCircleOIcon from '@ncigdc/theme/icons/CheckCircleOIcon';
```
Comment (caravinci) [inline src/packages/@ncigdc/modern_components/SuggestionFacet/SuggestionFacet.js]:
```suggestion
import CheckCircleOIcon from '@ncigdc/theme/icons/CheckCircleOIcon';
import { IRawQuery } from '@ncigdc/utils/uri/types';
```

File: src/packages/@ncigdc/components/Aggregations/SuggestionFacet.js [hunk 9]
```diff
@@ -23,7 +23,6 @@ import styled from '@ncigdc/theme/styled';
 import { dropdown } from '@ncigdc/theme/mixins';
 import Link from '@ncigdc/components/Links/Link';
 import CheckCircleOIcon from '@ncigdc/theme/icons/CheckCircleOIcon';
```
Comment (caravinci) [inline src/packages/@ncigdc/components/Aggregations/SuggestionFacet.js]:
```suggestion
import CheckCircleOIcon from '@ncigdc/theme/icons/CheckCircleOIcon';
import { IRawQuery } from '@ncigdc/utils/uri/types';
```

**Pull request merged**